devito/function.py

```diff
@@ -1033,15 +1042,43 @@ def _arg_check(self, args, intervals):
     _pickle_kwargs = Function._pickle_kwargs + ['time_order', 'save']
 
 
+class SubFunction(Function):
+    """
+    A :class:`Function` that is bound to another "parent" TensorFunction.
+
+    A SubFunction hands control of argument binding and halo exchange to its
+    parent TensorFunction.
+    """
+
+    def __init__(self, *args, **kwargs):
+        if not self._cached():
+            super(SubFunction, self).__init__(*args, **kwargs)
+            self._parent = kwargs['parent']
+
+    def _halo_exchange(self):
+        return
+
+    def _arg_values(self, **kwargs):
+        if self.name in kwargs:
+            raise RuntimeError("`%s` is a SubFunction, so it can't be assigned "
+                               "a value dynamically" % self.name)
+        else:
+            return self._parent._arg_defaults(alias=self._parent).reduce_all()
+
+
 class AbstractSparseFunction(TensorFunction):
     """
     An abstract class to define behaviours common to any kind of sparse
     functions, whether using precomputed coefficients or computing them
     on the fly. This is an internal class only and should never be
     instantiated.
     """
-    # Symbols that are encapsulated within this symbol (e.g. coordinates)
-    _child_functions = []
+
+    _sparse_position = -1
```

Comment:
Bit hacky no?

Reply:
Yes I know, just saying hard_coding index position seems a bit dangerous, for time too, don't have a solution just food for thoughts